choreo-extension-native/src/main/java/io/ballerina/observe/choreo/StepCountObserver.java

```diff
@@ -0,0 +1,90 @@
+ * limitations under the License.
+ */
+
+package io.ballerina.observe.choreo;
+
+import io.ballerina.observe.choreo.logging.LogFactory;
+import io.ballerina.observe.choreo.logging.Logger;
+import io.ballerina.runtime.observability.BallerinaObserver;
+import io.ballerina.runtime.observability.ObserverContext;
+import io.ballerina.runtime.observability.metrics.DefaultMetricRegistry;
+import io.ballerina.runtime.observability.metrics.MetricId;
+import io.ballerina.runtime.observability.metrics.MetricRegistry;
+import io.ballerina.runtime.observability.metrics.Tag;
+
+import java.util.Set;
+
+/**
+ * Observe the runtime and collect measurements.
+ */
+public class StepCountObserver implements BallerinaObserver {
+    private static final String PROPERTY_START_TIME = "_choreo_observation_start_time_";
+
+    private static final Logger LOGGER = LogFactory.getLogger();
+
+    private static long stepTime;
+
+    static {
+    String propertysteptime = System.getenv("CHOREO_PER_STEP_TIME");
+    if (propertysteptime == null) { 
+                propertysteptime = "500";
+            }
+    stepTime = Long.parseLong(propertysteptime);
+    }
+
+    @Override
+    public void startServerObservation(ObserverContext observerContext) {
+        startObservation(observerContext);
+    }
+    @Override
+    public void startClientObservation(ObserverContext observerContext) {
+        startObservation(observerContext);
+    }
+    @Override
+    public void stopServerObservation(ObserverContext observerContext) {
+        if (!observerContext.isStarted()) {
+            // Do not collect metrics if the observation hasn't started
+            return;
+        }
+        stopObservation(observerContext);
+    }
+    @Override
+    public void stopClientObservation(ObserverContext observerContext) {
+        if (!observerContext.isStarted()) {
+            // Do not collect metrics if the observation hasn't started
+            return;
+        }
+        stopObservation(observerContext);
+    }
+    
+    private void startObservation(ObserverContext observerContext) {
+        observerContext.addProperty(PROPERTY_START_TIME, System.currentTimeMillis());
+    }
+    
+    private void stopObservation(ObserverContext observerContext) {
+        Set<Tag> tags = observerContext.getAllTags();
+        try {
+            Long startTime = (Long) observerContext.getProperty(PROPERTY_START_TIME);
+            long duration = System.currentTimeMillis() - startTime;
+            long steps = Math.round(Math.ceil((double) (duration - stepTime) / stepTime));
+            final MetricRegistry metricRegistry = DefaultMetricRegistry.getInstance();
+            metricRegistry.counter(new MetricId("choreo_steps_total", "Total no of steps", tags)).increment(steps);
+        } catch (RuntimeException e) {
+            LOGGER.error("error collecting metrics for multiple metrics with tags " 
+            + tags + ": " + e.getMessage());
```

Comment:
Generally we keep a tab when we go into multiple lines to indicate that it is from the previous line. It makes it easier to read when there is a lot of code.
```suggestion
                + tags + ": " + e.getMessage());
```